### DISCUSSION TITLE
Getting `Cannot read properties of undefined (reading 'http')` error while setting up custom chain

### DISCUSSION THREAD

**Discussion by roudra323** [rainbow-me/rainbowkit#1576] [Q&A]
This is the code for setting up the custom chain (hardhat local node) :
```javascript
import "./polyfills";
import "./global.css";
import React from "react";
import ReactDOM from "react-dom/client";
import App from "./App.jsx";
import "./index.css";
import "@rainbow-me/rainbowkit/styles.css";
import { publicProvider } from "wagmi/providers/public";
import { jsonRpcProvider } from "wagmi/providers/jsonRpc";

import {
  getDefaultWallets,
  RainbowKitProvider,
  darkTheme,
} from "@rainbow-me/rainbowkit";
import { configureChains, createConfig, sepolia, WagmiConfig } from "wagmi";
import { polygon, optimism, arbitrum, zora, mainnet } from "wagmi/chains";
import { GlobalProvider } from "./context";

const localNode = {
  id: 43114,
  name: "Local Node",
  network: "ethereum",
  iconBackground: "#fff",
  nativeCurrency: {
    decimals: 18,
    name: "Ether",
    symbol: "Eth",
  },
  rpcUrls: {
    // Update this URL to your Hardhat local node URL
    default: { http: ["http://127.0.0.1:8545/"] },
  },
  testnet: false,
};

const { providers, Chain } = configureChains([localNode], [publicProvider()]);

const { connectors } = getDefaultWallets({
  appName: "vite_starter_pack",
  projectId: import.meta.env.VITE_RAINBOWKIT_KEY,
  Chain,
});

const wagmiConfig = createConfig({
  autoConnect: true,
  connectors,
  providers,
});

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <WagmiConfig config={wagmiConfig}>
      <RainbowKitProvider
        chains={Chain}
        theme={darkTheme({
          accentColor: "#7b3fe4",
          accentColorForeground: "white",
          borderRadius: "small",
          fontStack: "system",
          overlayBlur: "small",
        })}
      >
        <GlobalProvider>
          <App />
        </GlobalProvider>
      </RainbowKitProvider>
    </WagmiConfig>
  </React.StrictMode>
);
```
And in the browser I am getting the error in the console:
```
public.js:6 Uncaught TypeError: Cannot read properties of undefined (reading 'http')
    at public.js:6:31
    at configureChains (chunk-TSH6VVF4.js:29:25)
    at main.jsx:37:30
  
```

**Comment by DanielSinclair** [comment]
@roudra323 Can you share a CodeSandbox example? This could be related to your project configuration and polyfills, or other dependency issues.

**Reply by roudra323** [reply]
Here is the [CodeSandbox ](https://codesandbox.io/p/github/roudra323/vite_dapp_starter_template/main?import=true&layout=%257B%2522sidebarPanel%2522%253A%2522SEARCH%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clo8md8gz0007356d647tuhwh%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clo8md8gy0003356dwh155h9m%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clo8md8gy0005356dgevqkvf4%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clo8md8gy0006356dmi930mfy%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B60%252C40%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clo8md8gy0003356dwh155h9m%2522%253A%257B%2522id%2522%253A%2522clo8md8gy0003356dwh155h9m%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clo8md8gy0002356dmrr7mfv8%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.gitignore%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clo8md8gy0002356dmrr7mfv8%2522%257D%252C%2522clo8md8gy0006356dmi930mfy%2522%253A%257B%2522id%2522%253A%2522clo8md8gy0006356dmi930mfy%2522%252C%2522activeTabId%2522%253A%2522clo8n07i600zj356d163dpdvj%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522ENV_SETUP%2522%252C%2522id%2522%253A%2522clo8md8ln000p356dmnk1a7qy%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clo8n07i600zj356d163dpdvj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clo8md8gy0005356dgevqkvf4%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clo8md8gy0004356djhd5wzhm%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clo8mzt6c0003eejk8pscdxmf%2522%257D%255D%252C%2522id%2522%253A%2522clo8md8gy0005356dgevqkvf4%2522%252C%2522activeTabId%2522%253A%2522clo8md8gy0004356djhd5wzhm%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)example. In the  CodeSandBox I am getting `Service Worker registration failed` error . But if you restart the vite server by pressing `R` then you can see my mentioned `http` error.

**Comment by launchthatbrand** [comment]
I am getting the same error. Any solutions? I downgraded to wagmi v1 to work with rainbowkit as its not updated to work with wagmi v2, and still cant get it to work

**Reply by DanielSinclair** [reply]
@launchthatbrand This sounds like a Vite-specific issue, but reference the Wagmi v2 migration guide here: https://beta.rainbowkit.com/guides/rainbowkit-wagmi-v2
